src/components/OntologyBrowser/store.ts

```diff
@@ -212,6 +123,32 @@ export class PhenotypeStore {
     }
   };
 
+  getTreeNodeForKey = (key: string, treeNode = this.tree): TreeNode | null => {
+    let result: TreeNode | null = null;
+    for (let i = 0; i < treeNode.length; i++) {
+      if (treeNode[i].key === key) {
+        result = treeNode[i];
+        break;
+      }
+      result = this.getTreeNodeForKey(key, treeNode[i].children);
+      if (result) {
+        break;
+      }
+    }
+    return result;
+  };
+
+  getChildrenKeys = (node: TreeNode, root = false) => {
+    let nKeys: string[] = [];
+    node.children.forEach((i) => {
```

Comment:
why not using map?

Reply:
map wont work, but flapmap can. I'll run some benchmark to see what is optimal